pkg/sdk/urls.go

```diff
@@ -1,10 +1,33 @@
 package sdk
 
-import "fmt"
+import (
+	"fmt"
+	"net/url"
+)
 
-// HostToWebsocketURL converts the provided host into a websocket URL that can
+// RPCToWebsocketURL converts the provided URL into a websocket URL string that can
 // be used to subscribe to onchain events and query the chain via a client
 // context or send transactions via a tx client context.
-func HostToWebsocketURL(host string) string {
-	return fmt.Sprintf("ws://%s/websocket", host)
+func RPCToWebsocketURL(hostUrl *url.URL) string {
+	if hostUrl.Scheme == "http" {
+		return fmt.Sprintf("ws://%s/websocket", hostUrl.Host)
+	} else if hostUrl.Scheme == "ws" {
+		return fmt.Sprintf("ws://%s/websocket", hostUrl.Host)
+	}
+
+	return fmt.Sprintf("wss://%s/websocket", hostUrl.Host)
+}
+
+// ConstructGRPCUrl constructs a gRPC url string ensuring it contains either the scheme "grpcs" or "grpc"
+// This allows the SDK client control whether a TLS-enabled connection is used and some flexibility when specifying the gRPC URL.
+func ConstructGRPCUrl(hostUrl *url.URL) string {
+	if hostUrl.Scheme == "http" {
+		return fmt.Sprintf("grpc://%s", hostUrl.Host)
+	} else if hostUrl.Scheme == "grpc" {
+		return fmt.Sprintf("grpc://%s", hostUrl.Host)
+	} else if hostUrl.Scheme == "tcp" {
+		return fmt.Sprintf("tcp://%s", hostUrl.Host)
+	}
+
+	return fmt.Sprintf("grpcs://%s", hostUrl.Host)
```

Comment:
I've never seen `grpcs` or `grpc` be used anywhere as a scheme in the past. Can you link to something showing that this is correct and/or a standard?

My goal here is to:
1. Avoid setting new standards.	
2. My understanding is that gRPC is built on top of http2, which uses TLS 100% of the time.

For reference, here's a quick GPT response but googling online yields to the same endpoint:

![Screenshot 2024-04-01 at 2 38 49 PM](https://github.com/pokt-network/poktroll/assets/1892194/570f7757-7a81-47bb-824a-f904a8412653)